### PR TITLE
customized word tokenizer

### PR DESCRIPTION
**Pull request by SoluMilken**
@Liangtaiwan  @noobOriented 
## Word Tokenizer Given User Words
```python
E.g.
>>> from uttut.pipeline.ops.custom_word_tokenizer import CustomWordTokenizer
>>> op = CustomWordTokenizer(['珍奶', '珍奶去冰', '去冰'])
>>> output_seq, label_aligner = op.transform("一杯珍奶去冰")
>>> output_labels = label_aligner.transform([1, 2, 3, 3, 3, 0])
>>> output_seq
["一", "杯", "珍奶去冰"]
>>> output_labels
[1, 2, 3]
>>> label_aligner.inverse_transform(output_labels)
[1, 2, 3, 3, 3, 3]
```

### CODE REVIEW

File: uttut/pipeline/ops/utils/trie.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        for char in word:
+            if char not in current.children:
+                current.children[char] = TrieNode()
+            current = current.children[char]
+        current.word = word
+
+    def match_prefix(self, word: str, shortest: bool = False, start_idx: int = 0):
+        current = self.root
+        longest_word = None
+        for idx in range(start_idx, len(word)):
```
Comment (noobOriented) [inline uttut/pipeline/ops/utils/trie.py]: `for c in word[start_idx:]:`

Comment (SoluMilken) [reply uttut/pipeline/ops/utils/trie.py]: 所以 就不要用  start_idx 了

File: uttut/pipeline/ops/utils/trie.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    def has_child(self, key):
+        return key in self._children
+
+    def get_child(self, key):
+        return self._children[key]
+
+    def insert_child(self, key):
+        self._children[key] = TrieNode()
+
+    def set_word(self, word):
```
Comment (noobOriented) [inline uttut/pipeline/ops/utils/trie.py]:
```python
@property
def word(self):
    return self._word

@word.setter
def word(self, word):
    if not self._word:
        self._word = word
    else:
        raise ValueError()
```

Comment (SoluMilken) [reply uttut/pipeline/ops/utils/trie.py]: 32b7d7e

**Pull request merged**